lib/ansible/modules/packaging/os/yum.py

```
@@ -390,6 +390,14 @@ def __init__(self, module):
         self.pkg_mgr_name = "yum"
         self.lockfile = '/var/run/yum.pid'
 
+    def _enablerepos_with_error_checking(self, yumbase):
+        for rid in self.enablerepo:
+            try:
+                yumbase.repos.enableRepo(rid)
+            except yum.Errors.YumBaseError as e:
+                if u'repository not found' in to_text(e):
+                    self.module.fail_json(msg="Repository %s not found." % rid)
```

Did you intend to silently swallow all other YumBaseError exceptions here?